C/hal/Ultrasonic/ultrasonic.c

```diff
@@ -0,0 +1,43 @@
+#if MCU_TYPE == _AVR
```

Comment:
In `HAL` layer, I didn't expect to see handling of mcu types. In my opinion we shouldn't care about mcu type here. I know this checking is mandatory for handling delay in `avr`, So that will lead us to a result where delay system has to be improved. As you may know we have multiple ways to achieve delay:
1. using busy loop 
2. using `_delay_us` which is mapping to assembly instructions
3. using timer 
So I think the best place to do this is in service layer, where we have a wrapper function e.g. `dealy()` and inside this function we check either `pic` or `avr` and for the way the user want to use `BUILT_IN_FUNCT` `BUSY_LOOP` `TIMER_DELAY`

You can start in this implementation in separate PR